tests/linalg_test.py

```diff
@@ -1684,9 +1684,11 @@ def osp_fun(A):
     pivoting=[False, True]
   )
   def testScipyQrModes(self, shape, dtype, mode, pivoting):
```

Comment:
I figured it out! We need to add:

```
@jax.default_matmul_precision("float32")
```

Otherwise we lose precision on some hardware!

Reply:
Nice one! I've added this in and fixed the merge conflict.